dataland-frontend/src/components/forms/parts/elements/basic/UploadDocumentsForm.vue

```diff
@@ -102,21 +103,31 @@ export default defineComponent({
      * @param event full event object containing the files
      * @param event.files files
      */
-    async handleFilesSelected(event: FileUploadSelectEvent): Promise<void> {
+    handleFilesSelected(event: FileUploadSelectEvent) {
       const selectedFilesByUser = event.files as File[];
       if (isThereActuallyANewFileSelected(selectedFilesByUser, this.documentsToUpload)) {
-        const lastSelectedFile = selectedFilesByUser[selectedFilesByUser.length - 1];
-        const documentToUpload = { file: lastSelectedFile } as DocumentToUpload;
-        documentToUpload.reference = await calculateSha256HashFromFile(documentToUpload.file);
-        documentToUpload.fileNameWithoutSuffix = removeFileTypeExtension(documentToUpload.file.name);
-        this.documentsToUpload.push(documentToUpload);
-        this.emitDocumentsChangedEvent();
+        const documentsToUpload = Promise.all(
+          selectedFilesByUser.map(async (file) => {
+            return {
+              file: file,
+              reference: await calculateSha256HashFromFile(file),
+              fileNameWithoutSuffix: removeFileTypeExtension(file.name),
+            };
+          }),
+        ) as Promise<DocumentToUpload[]>;
+
+        void documentsToUpload.then((documentsToUpload) => {
+          this.documentsToUpload = documentsToUpload;
+          this.emitDocumentsChangedEvent();
+        });
       }
     },
     /**
      * Emits event that selected documents changed
      */
     emitDocumentsChangedEvent() {
+      console.log("emitting", this.documentsToUpload);
+      this.$emit("reportsUpdated", this.documentsToUpload);
       this.$emit("documentsChanged", this.documentsToUpload);
```

Comment:
Why are there two emits now with different names, but the same output?

Reply:
Removed. Done.

Reply:
👍